python/az/aro/azext_aro/_validators.py

```diff
@@ -283,3 +289,74 @@ def validate_load_balancer_managed_outbound_ip_count(namespace):
     if namespace.load_balancer_managed_outbound_ip_count < minimum_managed_outbound_ips or namespace.load_balancer_managed_outbound_ip_count > maximum_managed_outbound_ips:  # pylint: disable=line-too-long
         error_msg = f"--load-balancer-managed-outbound-ip-count must be between {minimum_managed_outbound_ips} and {maximum_managed_outbound_ips} (inclusive)."  # pylint: disable=line-too-long
         raise InvalidArgumentValueError(error_msg)
+
+
+def validate_enable_managed_identity(namespace):
+    if not namespace.enable_managed_identity:
+        return
+
+    if namespace.client_id is not None:
+        raise InvalidArgumentValueError('Must not specify --client-id when --enable-managed-identity is True')
+
+    if namespace.client_secret is not None:
+        raise InvalidArgumentValueError('Must not specify --client-secret when --enable-managed-identity is True')
+
+    if namespace.version is None or not re.match(r'^[4-9]{1}\.[0-9]{1,2}\.[0-9]{1,2}$', namespace.version):
```

Comment:
The `if namespace.version is None` part of this conditional makes it so that the cx is going to have to explicitly pass a version even after the default install version is >= 4.14, but I don't think we have a way around enforcing this at the CLI level since the default install version is an "RP-side" thing - am I understanding correctly?

We could technically not enforce it at the CLI level and in doing so allow the request to hit the RP and fail from there, but I like this way you've done it so that a faulty request will fail as early as possible where possible.

Reply:
I think we'll need to push a dedicated CLI change after 4.14+ is the default, yeah. I don't see a better way around handling this since the CLI has no knowledge of what ARO's "default" install version is. 

I'm actually fine with removing our CLI-side validation if we want to just handle this in API static validation within the RP. I think that might be a better option.